user_modules/api/APITestDataGenerator.lua

```diff
@@ -113,7 +113,12 @@ local function getFloatValue(pTypeData)
     local max = pTypeData.maxvalue
     if not min then min = api.dataType.FLOAT.min end
     if not max then max = api.dataType.FLOAT.max end
-    value = tonumber(string.format('%.02f', math.random() + math.random(min, max-1)))
+    local random = math.random() + math.random(min, max-1)
+    if random > 0 then
+      value = tonumber(string.format('%.02f', random - math.abs(random)%0.01))
+    else
+      value = tonumber(string.format('%.02f', random + math.abs(random)%0.01))
+    end
```

Comment:
Any reason we don't just clamp the value? Seems like a more general solution
```suggestion
    value = tonumber(string.format('%.02f', math.min(math.max(random, min), max)))
```